### PR TITLE
docs: Describe remote user management webui

### PR DESCRIPTION
**Pull request by tara-hpe**
Describe how to enable remote user (and the groups they belong to) management via the master configuration file for both OIDC and SAML. Describe how to manage remote users and groups via webui. Update RBAC instructions. Introduce concept of remote authentication and remote user group management. Update master config ref description of the relevant parameters.

note: if [SAML functionality](https://github.com/determined-ai/determined/pull/8529) will not be released with 12/13 release then docs team will create a PR to update this one

### CODE REVIEW

File: docs/manage/users-remote.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+The first step is to configure the :ref:`master configuration file <master-config-reference>` to
+enable auto provisioning users and the remote management of any information attached to the users.
+
+-  Set ``oidc.auto_provision_users`` option to ``true`` in your :ref:`master configuration file
+   <master-config-reference>`.
+
+.. note::
+
+   If ``scim_enabled`` is ``true``, then ``oidc.auto_provision_users`` must be ``false``.
+
```
Comment (carolinaecalderon) [inline docs/manage/users-remote.rst:35]: If SCIM is enabled (meaning that `scim.enabled: true` in the config), then this overrides any OIDC configuration. For best practice, set `oidc.auto_provision_users: false` in this case.

File: docs/manage/users-remote.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+   oidc:
+       enabled: true
+       provider: "Okta"
+       idp_recipient_url: "https://determined.example.com"
+       idp_sso_url: "https://dev-00000000.okta.com"
+       client_id: "xx0xx0"
+       client_secret: "xx0xx0"
+       auto_provision_users: true
+
+Determined sets the username of the user to the IdP username. You cannot set the username
```
Comment (carolinaecalderon) [inline docs/manage/users-remote.rst:49]: Determined sets the username of the user to the IdP email.

File: docs/manage/users-remote.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+.. code::
+
+   {
+      username: dee.ray@example.com
+      groups: ['A', 'B'],
+      displayName: "Dee Ray"
+   }
+
+Each time the authenticated user accesses Determined, their information is passed to Determined, and
+the changes are made. For example, when a user is assigned to a new group via your IdP, that
```
Comment (carolinaecalderon) [inline docs/manage/users-remote.rst:85]: Each time the authenticated user accesses Determined, their information is passed to Determined, and
group memberships are updated.

**Review by carolinaecalderon — COMMENTED** [review]
Left some edit suggestions

**Review by hamidzr — APPROVED** [review]
rbac changes lgtm

File: docs/manage/users-remote.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+ Enable Remote User Management
+*******************************
+
+Set the Auto Provision Option
+=============================
+
+The first step is to configure the :ref:`master configuration file <master-config-reference>` to
+enable auto provisioning users and the remote management of any information attached to the users.
+
+-  Set ``oidc.auto_provision_users`` option to ``true`` in your :ref:`master configuration file
```
[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:29]: good catch! thank you!

[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:29]: resolved

File: docs/manage/users-remote.rst (hunk 2 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:49]: resolved

File: docs/manage/users-remote.rst (hunk 3 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:85]: resolved

File: docs/manage/users-remote.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst]: resolved

File: docs/manage/users-remote.rst [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+             idp_sso_url: "https://dev-00000000.okta.com"
+             client_id: "xx0xx0"
+             client_secret: "xx0xx0"
+             auto_provision_users: true
+             display_name_claim_name: "XYZ"
+
+   .. tab::
+
+      SAML
+
```
Comment (tara-det-ai) [inline docs/manage/users-remote.rst:60]: @salonig23 SAML section for review. if the info is correct, i will either leave it in the PR or move the content to its own PR depending on timing.

File: docs/manage/users-remote.rst [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+         saml:
+             enabled: true
+             provider: "Okta"
+             idp_recipient_url: "https://determined.example.com/saml/sso"
+             idp_sso_url: "https://myorg.okta.com/app/...sso/saml"
+             idp_cert_path: "okta.cert"
+             auto_provision_users: true
+             groups_attribute_name: "groups"
+             display_name_attribute_name: "disp_name"
+         security:
```
Comment (salonig23) [inline docs/manage/users-remote.rst:82]: RBAC is not required to autoprovision, but it is required to set the groups attribute name

File: docs/manage/users-remote.rst [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      .. code:: yaml
+
+         saml:
+             enabled: true
+             provider: "Okta"
+             idp_recipient_url: "https://determined.example.com/saml/sso"
+             idp_sso_url: "https://myorg.okta.com/app/...sso/saml"
+             idp_cert_path: "okta.cert"
+             auto_provision_users: true
+             groups_attribute_name: "groups"
```
Comment (salonig23) [inline docs/manage/users-remote.rst:80]: These two parameters are not required to enable auto-provisioning, they help sync group membership and display name respectively. If the code example is just for auto-provisioning then I would recommend removing these

File: docs/manage/users-remote.rst [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+Set the Groups Claim Name Option
+================================
+
+Determined receives OIDC and SAML claims via a JSON Web Token (JWT) that you send from your IdP. If
+there is a group that does not already exist in Determined, then the system creates the group.
+
+To enable group membership synchronization:
+
+-  Set the ``groups_claim_name`` option to match the claim name for group memberships from your
```
Comment (salonig23) [inline docs/manage/users-remote.rst:97]: The option is called `groups_claim_name` in OIDC, but `groups_attribute_name` in SAML

File: docs/reference/deploy/master-config-reference.rst [hunk 9]
```diff
@@ -1571,12 +1571,109 @@ The username for HTTP basic authentication (only allowed with ``type: basic``).
...
+
+   .. code:: yaml
+
+      saml:
+          enabled: true
+          provider: "Okta"
+          idp_recipient_url: "https://determined.example.com/saml/sso"
+          idp_sso_url: "https://myorg.okta.com/app/...sso/saml"
+          idp_cert_path: "okta.cert"
+          auto_provision_users: true
```
Comment (salonig23) [inline docs/reference/deploy/master-config-reference.rst:1674]: Could you add definitions for these options, similar to OIDC?

**Review by AmanuelAaron — APPROVED** [review]
`det user list` changes LGTM

File: docs/manage/users-remote.rst (hunk 6 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:82]: resolved

File: docs/manage/users-remote.rst (hunk 7 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:80]: resolved

File: docs/manage/users-remote.rst (hunk 8 above)
Comment (tara-det-ai) [reply docs/manage/users-remote.rst:97]: thank you!

File: docs/reference/deploy/master-config-reference.rst (hunk 9 above)
Comment (tara-det-ai) [reply docs/reference/deploy/master-config-reference.rst:1674]: @salonig23  added. please re-review.

Comment (salonig23) [reply docs/reference/deploy/master-config-reference.rst:1674]: Looks great, thank you!

**Review by salonig23 — APPROVED** [review]
SAML Changes LGTM!

**Pull request merged**